kratos/utilities/math_utils.h

```diff
@@ -713,19 +666,19 @@ class MathUtils
      * @param rSecondVector Second input vector
      * @return The resulting norm
      */
-    static inline TDataType Dot(
+    static inline double Dot(
         const Vector& rFirstVector,
         const Vector& rSecondVector
         )
     {
         Vector::const_iterator i = rFirstVector.begin();
         Vector::const_iterator j = rSecondVector.begin();
-        TDataType temp = TDataType();
+        double temp = double();
```

Comment:
Since this is a built-in type, I'd rather be explicit about the desired initial value:
```suggestion
        double temp = 0.0;
```